src/program.rs

```diff
@@ -122,23 +125,49 @@ impl Program {
         self.render()
     }
 
-    fn move_image(&mut self) -> Result<(), String> {
+    fn construct_dest_filepath(&self, src_path: &PathBuf) -> Result<PathBuf, String> {
         match std::fs::create_dir_all(&self.dest_folder) {
             Ok(_) => (),
             Err(e) => match e.kind() {
                 ErrorKind::AlreadyExists => (),
                 _ => return Err(e.to_string()),
             },
         };
+
+        let cur_filename = match src_path.file_name() {
+            Some(f) => f,
+            None => return Err("failed to read filename for current image".to_string()),
+        };
+        let newname = PathBuf::from(&self.dest_folder).join(cur_filename);
+        Ok(newname)
+    }
+
+    /// Copies currently rendered image to dest directory
+    /// TODO: Handle when file already exists in dest directory
+    fn copy_image(&mut self) -> Result<(), String> {
+        // Check if there are any images
+        if self.images.is_empty() {
+            return Err("No image to copy".to_string());
+        }
+        let opt = &fs_extra::file::CopyOptions::new();
+        let filepath = self.images.get(self.index).unwrap_or_else(|| {
+            panic!(format!(
+                "image index {} < max image index {}",
+                self.index,
+                self.images.len()
+            ))
+        });
+        let newname = self.construct_dest_filepath(filepath)?;
+        copy(filepath, newname, opt).map_err(|e| e.to_string())?;
+        self.render()
```

Comment:
I don't think we need to re-render here. Maybe just return `Ok(())` instead

Reply:
Okay

Reply:
This still needs to be fixed